### ISSUE TITLE
Fahrt von S1 wird angeboten, obwohl sie wegen einer Baustelle nicht hält - Thörishaus

### ISSUE THREAD

**Issue by Flurin-BLS** [openTdataCH/ojp-sdk#154]
SBB Fahrplan:
![Image](https://github.com/user-attachments/assets/4098fa13-8fcd-4db2-859d-1f831087db82)
![Image](https://github.com/user-attachments/assets/5353467d-03d4-433b-8177-08dfa29b8a22)

OJP 2.0:
https://tools.odpch.ch/ojp-demo-v2/search?from=8516161&to=8504115&mode_types=monomodal&transport_modes=public_transport&trip_datetime=2024-10-31%2021:51&stage=v2-test
![Image](https://github.com/user-attachments/assets/4278fa02-f621-470b-b20c-d49383f445c9)

BLS Mobil:
![Image](https://github.com/user-attachments/assets/cf2dbd26-b3d6-4379-a6b3-1273181bc494)

**Comment by Flurin-BLS** [comment]
Die Baustelle ist bei der BLS für die S1 erfasst:
![Image](https://github.com/user-attachments/assets/63d76bb2-f58a-4bb0-bd17-01599a9f88f6)

**Comment by Flurin-BLS** [comment]
auch spannend.. bei uns auf dem Web ist die Fahrt auch nicht drin...
![Image](https://github.com/user-attachments/assets/230aa7f8-a4f8-4874-81e3-8f7650856085)

**Comment by u233336** [comment]
Hallo zusammen

Ich glaube, wir kommen der Sache in Thörishaus Dorf langsam auf die Spur. Die Sache sieht wie folgt aus:

- Gemäss SOLL-Fahrplan legt die S1 zwischen Niederwangen und Flamatt nach 22 Uhr keinen Halt ein. In Thörishaus Dorf wird also gemäss SOLL-Fahrplan nicht gestoppt.
- Der täglich um ca. 04:00 Uhr geladene Tagesfahrplan (VDV454 REF-AUS) hingegen sendet einen Halt in Thörishaus Dorf. Diese Meldung kommt meines Wissens von der BLS.
- Gemäss dem Tagesfahrplan ist es korrekt, dass wir in OJP diesen Halt in Thörishaus Dorf anzeigen (wir zeigen korrekterweise aber keinen Halt in Thörishaus Station an, denn diese Station ist weder in den SOLL-Daten noch im Tagesfahrplan).
- Da der Onlinefahrplan (OFPL) scheinbar keinen Tagesfahrplan (VDV454 REF-AUS) erhält, zeigt die sbb.ch keinen Halt in Thörishaus Dorf an.

Es stellen sich mir nun die folgenden Fragen:

- Hält die S1 nach 22 Uhr nun effektiv in Thörishaus Dorf oder nicht?
- Weshalb sendet die BLS den Halt in Thörishaus Dorf? Ausser die S1 hält dort tatsächlich.
- Weshalb erhält / bezieht sbb.ch den Tagesfahrplan nicht?

Wir bleiben an diesem Thema selbstverständlich dran und informieren wieder. Aber kann die BLS in Erfahrung bringen, ob die S1 in Thörishaus Dorf nun effektiv hält oder nicht?

Gruss Christoph

**Comment by Flurin-BLS** [comment]
Ciao @u233336  - ich konnte die fragen bei uns besprechen... hier der aktuelle Stand:

- Hält die S1 nach 22 Uhr nun effektiv in Thörishaus Dorf oder nicht?
> - Nein, sie hält nicht.. und das Problem besteht auch heute --> SBB.ch und BLS.ch sind korrekt - OJP nicht


- Weshalb sendet die BLS den Halt in Thörishaus Dorf? Ausser die S1 hält dort tatsächlich.
> - Die Baustelle ist korrekt "eingeplant"
> - Wir wissen nicht wieso der Tagesfahrplan nicht korrekt geliefert wird - der Fehler besteht wirklich nur am aktuellen Betriebstag
 
- Weshalb erhält / bezieht sbb.ch den Tagesfahrplan nicht?
> - nehme an sie beziehen es auch, aber anders?

**Comment by u233336** [comment]
Hallo @Flurin-BLS  wir sind einen Schritt weiter: VDV454 REF-AUS meldet in Thörishaus Dorf "Durchfahrt = true", was ja korrekt ist. Aber diese Meldung kommt bei uns derzeit nicht an. Wir gehen dem nach und suchen den Fehler.

Also seitens BLS und seitens sbb.ch läuft alles korrekt. Wir arbeiten daran.

**Comment by Flurin-BLS** [comment]
Habt ihr mal eine SBB Baustelle angeschaut mit einer Haltestelle die am Abend ausfällt? Ist dort das Verhalten von OJP das gleiche?

**Comment by u233336** [comment]
Es gibt noch eine Herausforderung, nämlich die, dass Thörishaus Dorf im Tagesfahrplan geliefert wird, aber als "Durchfahrt" markiert ist. Zu klären nun, weshalb das Flag "Durchfahrt" nicht berücksichtigt wird.

**Comment by u233336** [comment]
2 Punkte zu tun:

1. mit CUS klären, weshalb in VDV454 REFAUS ein Halt in Thörishaus Dorf "produziert" wird (zwar mit Durchfahrt=TRUE), wenn diese Haltestelle gemäss SOLL-Fahrplan nicht angefahren wird weil Baustelle. Gemäss VDV-Schrift ist diese nicht ganz in Ordnung.
2. Feature erstellen, damit das VDV-Flag "Durchfahrt" in ZUkunft ausgewertet wird

**Comment by Flurin-BLS** [comment]
Merci fürs Update @u233336  - was heisst das konkret? Wie lange wird der Bug noch bestehen? Feature ist für OJP oder für CUS? Aus Abnehmersicht ist es einfach ein Bug, der sehr unschön ist und möglichst rasch gefixt werden sollte.

**Comment by u233336** [comment]
Hallo @Flurin-BLS 
Die Analysen laufen noch. Fakt ist: wir haben keine SOLL-Fahrten für den Bus 483 für heute und die kommenden Tage. Somit ist es auch ein Fehler bei der Dateneinlieferung. Weshalb wir im Tagesfahrplan die Fahrten bei sbb.ch haben und bei uns nicht, muss ich noch klären. Aber bitte auch mal bei der Busland nachfragen, weshalb diese Daten nicht in INFO+ sind

**Comment by tobiaszurbriggen** [comment]
@u233336 Ich nehme an deine letzte Antwort betrifft dieses Ticket: https://github.com/openTdataCH/ojp-sdk/issues/162
Und die Frage von @Flurin-BLS betrifft den Bug hier zur S1.

**Comment by u233336** [comment]
@tobiaszurbriggen Ja, sorry, mein Fehler! Mein Kommentar von 09:30 Uhr betrifft das Ticket um #162

**Comment by TO-mdv** [comment]
Meeting mit SBB, MENTZ Spezialisten zum Thema VDV454 Durchfahrt - wie dieses zu interpretieren ist.

**Comment by u233336** [comment]
Und hier die Analyse der Spezialisten bei Fahrplan und Echtzeit:

Die Ursache liegt in der Verarbeitung von gemäss ÄvT-Prozess in NeTS modifizierten Trassen durch CUS (ÄvT = Änderung verständigter Trassen).

Falls CUS von INFO+ eine modifizierte T-Variante (Sollfahrplan) übernimmt, lädt CUS als Basis für die weitere Verarbeitung deren G-Variante (Regelfahrplan) nach und legt die T-Variante darüber.
Dabei werden alle von der G-Variante bzw. vom Regelfahrplan abweichenden Fahrweg- und Haltepolitikänderungen als «ausserordentlich» ausgewiesen.

Zurzeit ist dieses Vorgehen nötig, damit Datenbezüger wie KIB auf Haltestellen mit ausserordentlicher Durchfahrt (unabhängig von der Fristigkeit der Anordnung) den Ausfall und ggf. Reisendenlenkungen publizieren können. 

Diese Verarbeitungslogik dürfte jedoch nicht in allen Fällen und nicht für alle Datenbezüger gleichermassen zielführend sein.
Im Zuge des Vorhabens «KI-Erkennungsschlüssel Ziel 2025» muss neben der von der Fristigkeit abhängigen Einkürzung des Zugziels bei Teilausfällen auch fachlich geklärt werden, wie analog dazu mit ausserordentlichen Durchfahrten und Halten umzugehen ist.

**Comment by u233336** [comment]
Seitens MENTZ bereits umgesetzt, wird aber in einem Major Release (EFA 11) umgesetzt
